vm/systemSmartContracts/staking_test.go

```diff
@@ -1 +1,448 @@
+	arguments.Arguments = []*big.Int{big.NewInt(100)}
+
+	retCode := stakingSmartContract.Execute(arguments)
+	assert.Equal(t, vmcommon.UserError, retCode)
+}
+
+func TestStakingSC_ExecuteFinalizeUnStake(t *testing.T) {
+	t.Parallel()
+
+	registrationData := stakingData{
+		StartNonce:    0,
+		Staked:        true,
+		UnStakedNonce: 10,
+		BlsPubKey:     nil,
+		StakeValue:    big.NewInt(100),
+	}
+	argument := big.NewInt(100)
+	stakeValue := big.NewInt(0)
+	marshalizedRegData, _ := json.Marshal(&registrationData)
+	eei, _ := NewVMContext(&mock.BlockChainHookStub{}, &mock.CryptoHookStub{})
+	eei.SetSCAddress([]byte("addr"))
+	eei.SetStorage([]byte(ownerKey), []byte("data"))
+	eei.SetStorage(argument.Bytes(), marshalizedRegData)
+	stakingSmartContract, _ := NewStakingSmartContract(stakeValue, eei)
+	arguments := CreateVmContractCallInput()
+	arguments.CallerAddr = []byte("data")
+	arguments.Function = "finalizeUnStake"
+	arguments.Arguments = []*big.Int{argument}
+
+	retCode := stakingSmartContract.Execute(arguments)
+	assert.Equal(t, vmcommon.Ok, retCode)
+
+	data := stakingSmartContract.eei.GetStorage(argument.Bytes())
+	assert.Equal(t, 0, len(data))
+
+	destinationBalance := stakingSmartContract.eei.GetBalance(arguments.CallerAddr)
+	senderBalance := stakingSmartContract.eei.GetBalance(argument.Bytes())
+	assert.Equal(t, big.NewInt(100), destinationBalance)
+	assert.Equal(t, big.NewInt(-100), senderBalance)
+}
+
+func TestStakingSC_ExecuteSlashOwnerAddrNotOkShouldErr(t *testing.T) {
+	t.Parallel()
+
+	stakeValue := big.NewInt(0)
+	eei := &mock.SystemEIStub{}
+	stakingSmartContract, _ := NewStakingSmartContract(stakeValue, eei)
+	arguments := CreateVmContractCallInput()
+	arguments.Function = "slash"
+
+	retCode := stakingSmartContract.Execute(arguments)
+	assert.Equal(t, vmcommon.UserError, retCode)
+}
+
+func TestStakingSC_ExecuteSlashArgumentsNotOkShouldErr(t *testing.T) {
+	t.Parallel()
+
+	stakeValue := big.NewInt(0)
+	eei := &mock.SystemEIStub{}
+	eei.GetStorageCalled = func(key []byte) []byte {
+		return []byte("data")
+	}
+	stakingSmartContract, _ := NewStakingSmartContract(stakeValue, eei)
+	arguments := CreateVmContractCallInput()
+	arguments.Function = "slash"
+	arguments.CallerAddr = []byte("data")
+
+	retCode := stakingSmartContract.Execute(arguments)
+	assert.Equal(t, vmcommon.UserError, retCode)
+}
+
+func TestStakingSC_ExecuteSlashUnmarhsalErr(t *testing.T) {
+	t.Parallel()
+
+	stakeValue := big.NewInt(0)
+	eei := &mock.SystemEIStub{}
+	eei.GetStorageCalled = func(key []byte) []byte {
+		return []byte("data")
+	}
+	stakingSmartContract, _ := NewStakingSmartContract(stakeValue, eei)
+	arguments := CreateVmContractCallInput()
+	arguments.Function = "slash"
+	arguments.CallerAddr = []byte("data")
+	arguments.Arguments = []*big.Int{big.NewInt(100), big.NewInt(100)}
+
+	retCode := stakingSmartContract.Execute(arguments)
+	assert.Equal(t, vmcommon.UserError, retCode)
+}
+
+func TestStakingSC_ExecuteSlash(t *testing.T) {
```

Comment:
Suggestion: write a test which verifies that only an address with a stake can be slashed, and an address that has no stake cannot be slashed.